src/VS.Web.CG.Mvc/Identity/IdentityGeneratorTemplateModelBuilder.cs

```diff
@@ -255,15 +255,38 @@ internal Type UserType
         // Returns true if there is an existing layout page.
         // Note: layoutFile & supportFileLocation will always have a value when this exits.
         //      supportFileLocation is rooted
-        private bool DetermineSupportFileLocation(out string supportFileLocation, out string layoutFile)
+        internal bool DetermineSupportFileLocation(out string supportFileLocation, out string layoutFile)
         {
             string projectDir = Path.GetDirectoryName(_projectContext.ProjectFullPath);
 
             if (!string.IsNullOrEmpty(_commandlineModel.Layout))
             {
-                supportFileLocation = Path.GetDirectoryName(_commandlineModel.Layout);
-                layoutFile = _commandlineModel.Layout;
-                return true;
+                if (_commandlineModel.Layout.StartsWith("~"))
+                {
+                    layoutFile = _commandlineModel.Layout.Substring(1);
+                }
+                else
+                {
+                    layoutFile = _commandlineModel.Layout;
+                }
+
+                while (!string.IsNullOrEmpty(layoutFile) &&
+                    (layoutFile[0] == '\\' || layoutFile[0] == '/'))
+                {
+                    layoutFile = layoutFile.Substring(1);
+                }
+
+                // if the input layout file path consists of only slashes (and possibly a lead ~), it'll be empty at this point.
+                // So we'll treat it as if no layout file was specified (handled below).
+                if (!string.IsNullOrEmpty(layoutFile))
+                {
+                    // normalize the path characters
+                    layoutFile = layoutFile.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
```

Comment:
You're correct about the lead slash, here's how that's dealt with:
- This method always sets up a layout file path without a lead slash. 
- _ViewStart.cshtml generation always adds a slash at the start. see https://github.com/aspnet/Scaffolding/blob/dev/src/VS.Web.CG.Mvc/Templates/Identity/Pages/_ViewStart.cshtml

Reply:
No, I meant doing `layoutFile.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);`  would replace all `/` with `\` and the string in the `_ViewStart` would be using `\` (Eg. `"Views\Shared\Layout"` instead of `"Views/Shared/Layout"`)

Reply:
Ah, I see what you mean now. I think I can take that out. With the other changes, it should not be necessary.

Reply:
>Actually, in that spot, we should probably always change \ to /. The existing file selector in VS always gives forward slashes, but on the dos command line, a user could easily give a path with backslashes.

That sounds good to me.

Reply:
After further thought, that replace is so that Path.GetDirectoryName() functions properly on any platform. Therefore, that'll have to stay, and I'll add code to replace `\` with `/` in the final support file location:
```
// normalize the path characters
layoutFile = layoutFile.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);

supportFileLocation = Path.GetDirectoryName(layoutFile).Replace("\\", "/");
```